modular_skyrat/modules/cellguns/code/mediguns.dm

```diff
@@ -66,10 +66,18 @@
 // Upgrade Kit
 /obj/item/device/custom_kit/medigun_fastcharge
 	name = "VeyMedical CWM-479 upgrade kit"
-	desc = "Upgrades the internal battery inside of the medigun, allowing for faster charging and a higher cell capacity. Any cells inside of the origingal medigun during the upgrade process will be lost!"
+	desc = "Upgrades the internal battery inside of the medigun, allowing for faster charging and a higher cell capacity. Requires the medigun's cells to be removed first!"
+	// don't tinker with a loaded (medi)gun. fool
 	from_obj = /obj/item/gun/energy/cell_loaded/medigun/standard
 	to_obj = /obj/item/gun/energy/cell_loaded/medigun/upgraded
 
+/obj/item/device/custom_kit/medigun_fastcharge/pre_convert_check(obj/target_obj, mob/user)
+	var/obj/item/gun/energy/cell_loaded/medigun/standard/our_medigun = target_obj
+	if(length(our_medigun.installedcells))
+		balloon_alert(user, "unload it first!")
+		return FALSE
+	return TRUE
```

Comment:
thank you for this 🙏

Reply:
no prob, i just saw the bit about how it just deletes cells and thought "yeah i could probably do something about this"